src/main/java/org/tuberlin/de/best_taxidriver/BestTaxidriver.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+
+import org.apache.flink.api.java.tuple.Tuple11;
+import org.tuberlin.de.read_data.Taxidrive;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+
+
+/**
+ * Created by gerrit on 23.10.15.
+ */
+public final class BestTaxidriver {
+
+    public static void main(String[] args) throws Exception {
+        // set up the execution environment
+        final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+        // load taxi data from csv-file
+        DataSet<Taxidrive> taxidrives = env.readCsvFile("data/sorted_data.csv")
+                .pojoType(Taxidrive.class,
+                        "taxiID",
+                        "licenseID",
+                        "pickup_datetime",
+                        "dropoff_datetime",
+                        "trip_time_in_secs",
+                        "trip_distance",
+                        "pickup_longitude",
+                        "pickup_latitude",
+                        "dropoff_longitude",
+                        "dropoff_latitude",
+                        "payment_type",
+                        "fare_amount",
+                        "surcharge",
+                        "mta_tax",
+                        "tip_amount",
+                        "tolls_amount",
+                        "total_amount");
+
+        DataSet<DriverStat> stats  = createDriverHighscores(taxidrives);
+
+        DataSet<Tuple11<String, Integer, Double, Double, Double, Double, Integer, Double, Double, Double, Double>> statsAsTuple =
+                stats.map(new MapFunction<DriverStat, Tuple11<String, Integer, Double, Double, Double, Double, Integer, Double, Double, Double, Double>>() {
+                    @Override
+                    public Tuple11<String, Integer, Double, Double, Double, Double, Integer, Double, Double, Double, Double> map(DriverStat stat) throws Exception {
+                        return stat.getStatAsTuple();
+                    }
+                });
+
+        /*
+        // sort by average total amount
+        statsAsTuple.sortPartition(10, Order.DESCENDING).first(10).print();
+        // sort by average trip distance
+        statsAsTuple.sortPartition(8, Order.DESCENDING).first(10).print();
+        // sort by average trip time (fastest)
+        statsAsTuple.sortPartition(7, Order.DESCENDING).first(10).print();
+        // driver with most trips
+        */
+        statsAsTuple.sortPartition(10, Order.DESCENDING)
+                .setParallelism(1)
+                .writeAsCsv("data/driverstats.csv");
+
+        // execute program
+        env.execute("Driver Highscore");
+
+        //addHeader("data/driverstats.csv",DriverStat.getCSVHeader());
+    }
+
+    private static void addHeader(String filepath, String csvHeader) {
+        RandomAccessFile f = null;
```

Comment:
You should saving the result back into our HDFS system. When using HDFS this entire method will not work. You should make the filepath for the result as an argument like I mentioned above.